### ISSUE TITLE
Spawning Configs - A Revamp

### ISSUE THREAD

**Issue by MokahTGS** [Alex-the-666/Ice_and_Fire#1093]
Currently spawning controls for this mod is haphazard at best and completely non-configurable at worst.  I think a redesign that gives more control over what spawns where that includes all the creatures with equal spawning rules.

For example, I can configure if dragons spawn in a dimension or not but not if they can just spawn freely throughout the world.  Same thing with Pixies and Cyclops.

We just need a clean way to control where creatures spawn and how, that is separate from the structures.  Allow mod pack makers to turn off structure generation but still have creatures populate the world.  Let us determine what biomes and what dimensions they will spawn.

**Comment by ghost** [comment]
they should include dimension spawning because especialy the stymphilian birds are annoying in other dimensions

**Comment by Morinor** [comment]
And plz make the safe zone work...
Itd be prtty cool if you actually had to travell to far lands to find these mythical creatures, instead of just spawning right next to them.

**Comment by kreezxil** [comment]
I would like to have the ability to disable creatures entirely with a boolean value.

while we're on the subject of spawn configs, I'd like to see that massive section broken up into a part per each type.

As far as game mechanics go, I'd also like the option to set wether a mob is worldgen, allday, night, day. That way I can implement things where teir 1 dragons spawn at night, myrmex's are worldgen, pixies are day, stymphillians are allday, etc. For the mobs that spawn in structures, they would respawn at or near where their structures were.

**Comment by PlatinumFox** [comment]
Having each entity broken up into their own category would be GREAT! its pretty hard to find stuff right now.
I would also like to ask for the ability to blacklist each creature from spawning in a certain dimension.
Ex: i have 3 known creatures spawning in a dimension i do not want them in (ID: -23). I was able to disable dragons spawning in this dimension thanks to the config option. but other mobs are still spawning even when i  mods like the bad mobs mod.

Maybe even add global option to disable all I/F creatures from spawning in a certain dimension. (or other places)

**Comment by Thethuthinnang** [comment]
Is it all right to ask, what is the status of the dangerous worldgen config? I have to echo the comment about safe zones; I came here to post about that very issue. (I'm using the latest release.) One of the things I love to do is explore the world I generate; I like patient hunting and long adventures to be part of my MC play. So, I'd like to have this safe zone for a certain radius around spawn for Ice and Fire mobs. Right now, dragon roosts keep spawning around 200 blocks of my spawn point, sometimes two nearly side-by-side, no matter how I configure that setting. I think 200 is the default setting, but not sure why it's not responding to any other numbers I've tried putting in (which are well within the maximum limit). Could someone look into this? Thank you so much for this mod and for your help.
